### PR TITLE
[DOC] mail: cleaning of inline class documentation.

### PR DESCRIPTION
**Pull request by tde-banana-odoo**

### CODE REVIEW

**Comment by xmo-odoo** [comment]
Maybe rebase this on more recent 8.0 and replace the triple single-quote by the more common triple-double-quote for the docstring? Also pep8 uses double quotes.

File: addons/mail/mail_thread.py [hunk 1]
```diff
@@ -59,52 +59,68 @@ def decode_header(message, header, separator=' '):
...
+
+    Inheriting classes are not required to implement any method, as the default
+    implementation will work for any model. However it is common to override at
+    least the ``message_new`` and ``message_update`` methods (calling ``super``)
+    to add model-specific behavior at creation and update of a thread when
+    processing incoming emails.
+
+    New field parameters:
+
+     - track_visibility: automatic tracking of the field value. The parameter
```
Comment (xmo-odoo) [inline addons/mail/mail_thread.py]: Indentation looks incorrect

File: addons/mail/mail_thread.py [hunk 2]
```diff
@@ -59,52 +59,68 @@ def decode_header(message, header, separator=' '):
...
+       - False: default value, no tracking done
+       - 'onchange': the changes are tracked, and the value is displayed in a
+         message each time it has changed
+       - 'always': the changes are tracked, and the value is displayed in a
+         message everytime any tracked field changes its value
+
+    Options:
+
+     - _mail_flat_thread=<bool>: if set to True (default value), all messages are
+     automatically attached to the first posted message on the resource, leading to
```
Comment (xmo-odoo) [inline addons/mail/mail_thread.py]: needs to be indented

File: addons/mail/mail_thread.py [hunk 3]
```diff
@@ -59,52 +59,68 @@ def decode_header(message, header, separator=' '):
...
+       to be formatted like
+
+        _track = {
+            'field': {
+                'module.subtype_xml': lambda self, cr, uid, obj, context=None: obj[state] == done,
+                'module.subtype_xml2': lambda self, cr, uid, obj, context=None: obj[state] != done,
+            }, 'field2': {
+                ...
+            },
+        }
```
Comment (xmo-odoo) [inline addons/mail/mail_thread.py]: code block is not going to be formatted correctly, needs to be indented with a `::` before or in a `.. code-block::` directive

File: addons/mail/mail_thread.py [hunk 4]
```diff
@@ -59,52 +59,68 @@ def decode_header(message, header, separator=' '):
...
+
+        where
+        :param string field: field name. Each field present in the dict should
+                            have the track_visiblity attribute set (see above).
+        :param module.subtype_xml: xml_id of a mail.message.subtype (i.e. mail.mt_comment)
+        :param function lambda: returns whether the tracking should trigger the
+                                subtype, according to a condition
+        :param obj: is the browse_record of the current document on which the
+                    condition will be evaluated
```
Comment (xmo-odoo) [inline addons/mail/mail_thread.py]: `:param:` are for function parameter fields in a function docstring, I don't think their use here is correct. Maybe use a definition list instead?

**Comment by rim-odoo** [comment]
@tde-banana-odoo Don't leave this branch on her own

**Pull request closed without merging**